### PR TITLE
Nick's danebook

### PR DESCRIPTION
**Pull request by sicknarlo**

### CODE REVIEW

File: spec/features/users_spec.rb [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+  scenario "guests are brought to sign up page" do
+    expect(page).to have_content "Connect with all your friends!"
+  end
+
+  scenario "guests cannot sign in without entering a proper login and email" do
+    click_button "Sign In"
+    expect(page).to have_content "Connect with all your friends!"
+  end
+
+  scenario "guests can sign up" do
```
Comment (shadefinale) [inline spec/features/users_spec.rb:17]: Woah..... I'd refactor the filling out of forms in these following tests into a macro.

Consider a method that takes **args or an options hash, and fills in forms with either args[:option] || "my_default_option" for each form field.

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -0,0 +1,134 @@
+require 'rails_helper'
+
+describe User do
+  let(:user){build(:user)}
+
+  it "should be valid with default attributes" do
```
Comment (shadefinale) [inline spec/models/user_spec.rb:6]: I really like your thorough testing for the bounds of your user attribute validations.

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    @user = User.find(params[:id])
+  end
+
+  # Never trust parameters from the scary internet, only allow the white list through.
+  def user_params
+    params.require(:user).permit(:id, :first_name, :last_name, :email, :password, :password_confirmation, :birthday, :gender)
+  end
+
+  def friends_with?(user)
+    self.friendships.where(:friend_id => user.id)
```
Comment (facingsouth) [inline app/controllers/users_controller.rb:53]: This is might always return something truthy, where clause will return [] when they are not friends, but [] is truthy.

File: app/helpers/friendships_helper.rb [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    else
+      show_friend(user)
+    end
+  end
+
+  def show_unfriend(user)
+
+    link_to("Remove Friend", friendship_path(:friend_id => user),
+                             :method => :delete,
+                             class: "btn btn-primary pull-right")
```
Comment (facingsouth) [inline app/helpers/friendships_helper.rb]: You might want to add .html_safe at the end.

File: app/controllers/users_controller.rb (hunk 3 above)
Comment (facingsouth) [inline app/controllers/users_controller.rb:53]: You have this friends_with? method in both user model and users controller.

**Pull request closed without merging**